core/src/main/scala/kafka/server/metadata/ZkMetadataCache.scala

```diff
@@ -350,11 +347,7 @@ class ZkMetadataCache(
 
   override def getAliveBrokerNode(brokerId: Int, listenerName: ListenerName): Option[Node] = {
     val snapshot = metadataSnapshot
-    brokerId match {
-      case id if snapshot.controllerId.filter(_.isInstanceOf[KRaftCachedControllerId]).exists(_.id == id) =>
-        kraftControllerNodeMap.get(id)
-      case _ => snapshot.aliveBrokers.get(brokerId).flatMap(_.getNode(listenerName))
-    }
+    snapshot.aliveBrokers.get(brokerId).flatMap(_.getNode(listenerName))
```

Comment:
Interesting. Do you know why this was added in the first place? @akhileshchg

Reply:
I recall why we did this now (I didn't originally)...

During ZK migration, we wanted a way to make the controller node provider work with both ZK and KRaft controllers.

The correct way to do this is just to consult `RaftManager` for the endpoint when we're dealing with a KRaft controller (which we do know)

But at the time, we "hacked" it by adding the controllers to the cache as "brokers" for some functions but not others (ew)